pkg/frontend/transport/roundtripper.go

```diff
@@ -39,7 +42,20 @@ func (a *grpcRoundTripperAdapter) RoundTrip(r *http.Request) (*http.Response, er
 		return nil, err
 	}
 
-	resp, err := a.roundTripper.RoundTripGRPC(r.Context(), req)
+	var (
+		resp      *httpgrpc.HTTPResponse
+		reqValues url.Values
+		ts        time.Time
+	)
+
+	if strings.HasSuffix(r.URL.Path, "/query") || strings.HasSuffix(r.URL.Path, "/query_range") {
+		if err = r.ParseForm(); err == nil {
+			reqValues = r.Form
+			ts = time.Now()
+		}
+	}
+
+	resp, err = a.roundTripper.RoundTripGRPC(r.Context(), req, reqValues, ts)
```

Comment:
Is it better to have the `GetPriority` logic in QFE earlier? If we do it at `grpcRoundTripperAdapter`, then the input query has been split already (both vertically and horizontally).
It is better to me if we do it before any splitting. So that a single query can have a single query priority.

Reply:
+1

Reply:
Addressed in https://github.com/cortexproject/cortex/pull/5605/commits/aa38a8c556756af3337a3deb8537758cceaad585